### PR TITLE
Add bazel fossa deps support

### PR DESCRIPTION
**Pull request by cmboling**
Add support for converting a bazel dependency file into a `fossa-deps.yml`.

### CODE REVIEW

**Comment by cmboling** [comment]
Feel free to nitpick. This is just prototyping without the usual try/catch blocks and whatnot. Let me know if you find any edge cases.

@zlav @saramaebee how do we normally analyse only the `fossa-deps.yml`? Do we typically add a `.fossa.yml` or use the `--exclude-path` flag in `fossa analyze`?

File: python/scripts/bazel/fossa-deps/fossa-deps-converter.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    json.dump(dictionary, file)
+    file.close()
+
+def findGoRepostoryBlocks(input):
+    goRepositoryBlockRegex = re.compile(r"go_repository\([^)]*\)", re.IGNORECASE)
+    goRepositoryBlockMatches = goRepositoryBlockRegex.findall(input)
+    return goRepositoryBlockMatches
+
+def findImportPathValue(input):
+    importPathNameRegex = re.compile(r'importpath\s*=\s*"\s*(.*?)\s*"', re.IGNORECASE)
```
Comment (cmboling) [inline python/scripts/bazel/fossa-deps/fossa-deps-converter.py:21]: Regex patterns can be moved into `class _Const(object)` but it's whatevs

File: python/scripts/bazel/fossa-deps/fossa-deps-converter.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    bazelFile = readDepsBzl()
+    goRepositoryBlobs = findGoRepostoryBlocks(bazelFile)
+
+    referencedDepsDict = { "referenced-dependencies": [] }
+
+    for go_repo in goRepositoryBlobs:
+        importPath = findImportPathValue(go_repo)
+        dependencyVersion = findDependencyVersionValue(go_repo)
+
+        referencedDepsDict["referenced-dependencies"].append({"type":"go", "name":importPath, "version":dependencyVersion})
```
Comment (cmboling) [inline python/scripts/bazel/fossa-deps/fossa-deps-converter.py:40]: The script doesn't handle dupes, but idk if gazelle handles that or if it's something we need to worry about 🤔

File: python/scripts/bazel/fossa-deps/README [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# Bazel Dependency File Conversion
+
+Convert a Bazel dependency file to a `fossa-deps.yml`.
```
Comment (zlav) [inline python/scripts/bazel/fossa-deps/README:3]: I think you meant to say `fossa-deps.json` here and elsewhere in the README

**Comment by zlav** [comment]
>@zlav @saramaebee how do we normally analyze only the fossa-deps.yml? Do we typically add a .fossa.yml or use the --exclude-path flag in fossa analyze? or both? Does it matter or is this subjective

Its subjective, `--exclude-path` may be better so you can run `fossa-deps-converter.py && fossa analyze --exclude-path .` without worrying about the config file but at the end of the day its the same thing 🤷 

I don't have any comments on the implementation, this looks great and the results look great. Awesome README as well.

File: python/scripts/bazel/fossa-deps/README (hunk 3 above)
Comment (cmboling) [reply python/scripts/bazel/fossa-deps/README:3]: Ohhh good catch! 🪨 🚀

File: python/scripts/bazel/fossa-deps/fossa-deps-converter.py (hunk 1 above)
Comment (cmboling) [reply python/scripts/bazel/fossa-deps/fossa-deps-converter.py:21]: For later. Also the way it's written we're compiling the regex n times.

**Comment by cmboling** [comment]
Alrighty thanks Zach! Going to merge this.

**Pull request merged**